### ISSUE TITLE
unicode: U+0000 is not in Cc

### ISSUE THREAD

**Issue by jacobsa** [golang/go#10153]
The first line of [UnicodeData.txt][data] defines U+0000 as being in category Cc. But package unicode seems to [disagree][table], as confirmed by this program:

```Go
package main

import (
        "fmt"
        "unicode"
)

func main() {
        fmt.Println(unicode.In(0, unicode.Cc))
}
```

Given the provenance of this package and the fact that tables.go is auto-generated, I'm sure I've missed something. What's the reason for rune zero not being in `Cc`?

[data]: http://www.unicode.org/Public/7.0.0/ucd/UnicodeData.txt
[table]: https://github.com/golang/go/blob/master/src/unicode/tables.go#L82-L88

**Comment by ianlancetaylor** [comment]
CC @mpvl@golang.org

**Comment by rsc** [comment]
@robpike

This is probably due to unicode/maketables.go saying:

	if point == 0 {
		return // not interesting and we use 0 as unset
	}

Not clear how much we care. That's up to Rob and Marcel, but we should probably close this one way or the other.

**Comment by jacobsa** [comment]
Nice find. My opinion is that U+0000 should be in Cc, but it's only a weak and uninformed opinion. Mostly I'm curious about why it's regarded as uninteresting.

**Comment by mpvl** [comment]
Sorry, missed this earlier. I agree that U+0000 should be in Cc.  Not doing so will lead to bugs in certain Unicode algorithms.  OTOH, fixing it may expose bugs in faulty code quite easily.  Unless people object and want it earlier, I can fix this after the 1.5 freeze.  Ideally there is a long period before a freeze changing this.

**Comment by robpike** [comment]
Let's do it in 1.6. A freeze is a freeze. Arguably this is a bug but it's also a behavior change.
